### PR TITLE
[SM-223] Web: change `<h2>` modal title to `<h1>`, tweak styles to keep same look

### PR DESCRIPTION
**Pull request by patrickhlauke**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Change primary heading in modals from `<h2>` to `<h1>` - for consistency with the desktop app modals, and because logically in a modal (which is its own little document) the top-most heading is the primary heading. Not a critical issue, but mostly good practice. See https://github.com/bitwarden/clients/pull/3323#discussion_r947780258

## Code changes

- **apps\web\src\app\ ... .component.html**: swap out `<h2 class="modal-title">` for `<h1 class="modal-title">` in the modal dialogs
- **apps\web\src\scss\modals.scss**: add extra styling to keep the look and feel of the modals the same (set the same explicit `font-size` for `.modal-title` as regular `h2`s)

## Screenshots

No change

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

**Comment by djsmith85** [comment]
@danielleflinn Could you please review/approve the styling changes done in `apps\web\src\scss\modals.scss`. 
Would we prefer the normal size for `h1`'s or go with the modified style?

**Comment by danielleflinn** [comment]
h1s look great and I agree with the style tweaks Patrick made to mimic existing header styles. 

@patrickhlauke could you also update the `h2` in [this file](https://github.com/bitwarden/clients/blob/master/libs/components/src/dialog/dialog/dialog.component.html) to an `h1`?

**Comment by patrickhlauke** [comment]
@danielleflinn sure. should I be concerned that looking at that file, I don't see any `role="dialog"` etc? how/where is this used?

**Comment by danielleflinn** [comment]
@patrickhlauke yes we will need to add `role=dialog` and other aria attributes typically used in dialogs to that component as well; good catch! If you'd like to while you are in there feel free to add these, otherwise I'll have someone internally update it. 

That component is currently not in use but we have plans to update all modals in the web vault to utilize it as their base (that way to make 1 style or html tag change, we can do it in 1 place rather than the 40+ places you had to update the h2 in this PR). 

Updating the h2 should not effect the styles for the linked component as it uses Tailwind utility classes rather than a default html heading style. 

Hope this clarifies things! Let me know if you have additional thoughts

**Comment by patrickhlauke** [comment]
> @patrickhlauke yes we will need to add role=dialog and other aria attributes typically used in dialogs to that component as well; good catch! If you'd like to while you are in there feel free to add these, otherwise I'll have someone internally update it.

as I'm not too hot on Angular, i'll add the `role` and `aria-modal`, but for `aria-labelledby`, it'll need some autogenerated unique `id`, and not sure what the best approach here will be - unless those modals will always only ever be shown one at a time, in which case can easily set a single general one hardcoded in the component itself...or go for an `aria-label` using the same text as that's injected into the heading.

tailwind...urgh. would not be my choice, but fair enough ;)

**Comment by danielleflinn** [comment]
Thanks Patrick; I'll work with the team to get a solution in for `aria-labeledby` vs `aria-label`. Thanks for your contribution.

**Review by danielleflinn — APPROVED** [review]

**Comment by patrickhlauke** [comment]
@danielleflinn done the `role`/`aria-modal` (also updated the simple-dialog component just in case). hoping that when transitioning to these new components over time any improvements/approaches we made recently won't fall by the wayside again (e.g. the notifications/toasts, which finally announce correctly in AT after my upstream fix to https://github.com/scttcper/ngx-toastr/pull/934)

File: libs/components/src/dialog/simple-dialog/simple-dialog.component.html [hunk 1]
```diff
@@ -1,14 +1,16 @@
 <div
+  role="dialog"
+  aria-modal="true"
```
Comment (Hinton) [inline libs/components/src/dialog/simple-dialog/simple-dialog.component.html:3]: These gets set by the dialog service on the container.

File: libs/components/src/dialog/dialog/dialog.component.html [hunk 2]
```diff
@@ -1,13 +1,15 @@
 <div
+  role="dialog"
+  aria-modal="true"
```
Comment (Hinton) [inline libs/components/src/dialog/dialog/dialog.component.html:3]: These gets set by the dialog service on the container.

**Comment by Hinton** [comment]
@patrickhlauke I've changed the titles to h1 in https://github.com/bitwarden/clients/pull/3439 along with fixing labelledby. To avoid conflicts should we remove those changes from this PR?

**Comment by Hinton** [comment]
@djsmith85 from a code perspective this looks good once the component library dialogs have been updated. Could you look into checking with QA when they have some capacity to test this?

File: libs/components/src/dialog/dialog/dialog.component.html (hunk 2 above)
Comment (patrickhlauke) [reply libs/components/src/dialog/dialog/dialog.component.html:3]: ah, fair enough...should i remove this part?

Comment (Hinton) [reply libs/components/src/dialog/dialog/dialog.component.html:3]: Yes please.

**Comment by patrickhlauke** [comment]
@Hinton 
> Demo https://62a88a6de5b807fa98886113-mjjgrsvchd.chromatic.com/?path=/story/component-library-dialogs-service--default

That looks quite good. Note the "X" close control lacks an accessible name though (could do with an `aria-label="Close"` on the button)

File: libs/components/src/dialog/dialog/dialog.component.html (hunk 2 above)
Comment (patrickhlauke) [reply libs/components/src/dialog/dialog/dialog.component.html]: done

File: libs/components/src/dialog/simple-dialog/simple-dialog.component.html (hunk 1 above)
Comment (patrickhlauke) [reply libs/components/src/dialog/simple-dialog/simple-dialog.component.html]: removed

**Comment by patrickhlauke** [comment]
> @patrickhlauke I've changed the titles to h1 in #3439 along with fixing labelledby for the new dialogs. To avoid conflicts should we remove those changes from this PR?

I've just copied those changed headings from #3439 verbatim into this PR now, so shouldn't cause any conflicts (unless by doing so I messed things up somehow)

**Comment by Hinton** [comment]
Thanks for the feedback on the close button. I added both `title` and `aria-label`.

**Comment by Hinton** [comment]
I think it should work, we'll just have to merge #3439 first since the dialogs will stop working until both are merged.

**Review by Hinton — DISMISSED** [review]
Code change looks good.

**Comment by djsmith85** [comment]
@Hinton As discussed please continue championing this.

**Comment by Hinton** [comment]
@patrickhlauke I've merged #3439, could you please resolve the merge conflicts?

**Comment by patrickhlauke** [comment]
@Hinton done, chief

**Review by Hinton — APPROVED** [review]

**Comment by Hinton** [comment]
Merged, thanks @patrickhlauke.

**Pull request merged**